### PR TITLE
New Safereach prebid adapter

### PR DESCRIPTION
**Pull request by SafereachPrebid**
## Type of change

- [x] New bidder adapter   

- test parameters for validating bids
```
{
   	bidder: 'safereach',
   	params: {
      		placementId: 0,
      		traffic: 'banner'
    	}
}
```

- contact email of the adapter’s maintainer: safereachprebid@ukr.net
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/: https://github.com/prebid/prebid.github.io/pull/1080

### CODE REVIEW

**Comment by SafereachPrebid** [comment]
update

**Comment by SafereachPrebid** [comment]
@mkendall07 does it takes some time to make review or there is some problem ?

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging a2eb1efcf01a4038396873be6c92b73f2414d999 into 7b792a1e89814afc13b1897f91bdc2d0bc6149e2 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-addf086a3b7c8d77b081369a040cae6292bb1476)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: modules/safereachBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {object} bid The bid to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: (bid) => {
+    return Boolean(bid.bidId &&
+        bid.params &&
+        !isNaN(bid.params.placementId) &&
+        spec.supportedMediaTypes.indexOf(bid.params.traffic) !== -1
```
[the remark this answers is not in the document]
Comment (SafereachPrebid) [reply modules/safereachBidAdapter.js]: @mkendall07

**Comment by SafereachPrebid** [comment]
@mkendall07 Looks like there is some issues on circleci platform 

```
#!/bin/bash -eo pipefail
# Download the browserstack binary file
wget "https://www.browserstack.com/browserstack-local/BrowserStackLocal-linux-x64.zip"
# Unzip it
unzip BrowserStackLocal-linux-x64.zip
# Run the file with user's access key
./BrowserStackLocal ${BROWSERSTACK_ACCESS_KEY} &
--2019-01-31 07:44:02--  https://www.browserstack.com/browserstack-local/BrowserStackLocal-linux-x64.zip
Resolving www.browserstack.com (www.browserstack.com)... 52.21.30.192, 34.224.20.64, 23.20.105.117, ...
Connecting to www.browserstack.com (www.browserstack.com)|52.21.30.192|:443... connected.
HTTP request sent, awaiting response... 504 GATEWAY_TIMEOUT
2019-01-31 07:45:02 ERROR 504: GATEWAY_TIMEOUT.

Exited with code 8
```

File: modules/safereachBidAdapter.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Test Parameters
+```
+    var adUnits = [{
+                code: 'placementCode',
+                sizes: [[300, 250]],
+                bids: [{
+                        bidder: 'safereach',
+                        params: {
+                            placementId: 0,
+                            traffic: 'banner'
```
Comment (mkendall07) [inline modules/safereachBidAdapter.md]: I'm talking about this param. It looks like a "mediaType". What are the other possible values for this param?

Comment (SafereachPrebid) [reply modules/safereachBidAdapter.md]: @mkendall07 other values are native and video, according to this param rtb server handles request

**Pull request closed without merging**